synapse/metrics/background_process_metrics.py

```diff
@@ -197,7 +199,15 @@ def run():
                 _background_processes.setdefault(desc, set()).add(proc)
 
             try:
-                yield func(*args, **kwargs)
+                # We ensureDeferred here to handle coroutines
```

Comment:
I thought it was legit to `yield` on a coroutine?

Reply:
I get a `RuntimeWarning: coroutine 'ClientIpStore._prune_old_user_ips' was never awaited` if I don't do this

Reply:
that is... surprising. We certainly `yield` on async functions elsewhere.

Reply:
... but only where we have an ensureDeferred at the top of the call stack.